src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -35,47 +35,214 @@
 #include "application_manager/application_manager_impl.h"
+#include "application_manager/mock_resumption_data.h"
+#include "application_manager/resumption/resume_ctrl_impl.h"
 #include "application_manager/mock_application_manager_settings.h"
-#include "policy/mock_policy_settings.h"
+#include "hmi_message_handler/mock_hmi_message_handler.h"
+#include "connection_handler/mock_connection_handler.h"
+#include "protocol_handler/mock_session_observer.h"
+#include "policy/usage_statistics/mock_statistics_manager.h"
+
+namespace test {
+namespace components {
+namespace application_manager_test {
 
 namespace am = application_manager;
 namespace policy_test = test::components::policy_handler_test;
+namespace con_test = connection_handler_test;
+
+using testing::_;
 using ::testing::Return;
 using ::testing::ReturnRef;
-
-namespace test {
-namespace components {
-namespace application_manager_test {
+using ::testing::NiceMock;
+using ::testing::SetArgReferee;
+using ::testing::SetArgPointee;
+using ::testing::Truly;
+using ::testing::AtLeast;
+using ::testing::DoAll;
+using usage_statistics_test::MockStatisticsManager;
 
 namespace {
 const std::string kHmiCapabilitiesFile("hmi_capabilities.json");
 const uint32_t kThreadPoolSize = 1;
 const bool kHmiLaunch = false;
 const uint32_t kStopStreamingTimeout = 10000;
+const uint32_t kWaitFreeQueue = 2000;
+
+bool is_called_ = false;
+uint32_t hmi_app_id_ = 0u;
+sync_primitives::Lock state_lock_;
+sync_primitives::ConditionalVariable state_condition_;
 }  // namespace
 
 class ApplicationManagerImplTest : public ::testing::Test {
  protected:
   virtual void SetUp() OVERRIDE {
-    app_manager_impl.reset(new am::ApplicationManagerImpl(
-        mock_application_manager_settings_, mock_policy_settings_));
+    const uint32_t timeout = 10000u;
+    std::string directory_name = "./test_storage";
+    const std::string folder_name = "app";
+    const std::string test_path =
+        file_system::ConcatPath(directory_name, folder_name);
+    const std::string file1 =
+        file_system::ConcatPath(test_path, "test_file2.bmp");
+
+    ASSERT_TRUE(file_system::CreateDirectoryRecursively(test_path));
+    ASSERT_TRUE(file_system::CreateFile(file1));
+
+    ON_CALL(mock_application_manager_settings, thread_pool_size())
+        .WillByDefault(Return(2u));
+    ON_CALL(mock_application_manager_settings, app_icons_folder())
+        .WillByDefault(ReturnRef(directory_name));
+    EXPECT_CALL(mock_application_manager_settings, app_storage_folder())
+        .WillRepeatedly(ReturnRef(directory_name));
+    ON_CALL(mock_application_manager_settings, launch_hmi())
+        .WillByDefault(Return(true));
+    ON_CALL(mock_application_manager_settings, stop_streaming_timeout())
+        .WillByDefault(Return(1u));
+    ON_CALL(mock_application_manager_settings, default_timeout())
+        .WillByDefault(ReturnRef(timeout));
+    app_manager_impl_.reset(new am::ApplicationManagerImpl(
```

Comment:
@AGritsevich  is it possible to avoid mixing creating components creation anf ON_CALL?